File: code/go/ch21/src/luago/api/consts.go
package api

const LUA_MINSTACK = 20
const LUAI_MAXSTACK = 1000000
const LUA_REGISTRYINDEX = -LUAI_MAXSTACK - 1000
const LUA_RIDX_MAINTHREAD int64 = 1
const LUA_RIDX_GLOBALS int64 = 2
const LUA_MULTRET = -1

const (
	LUA_MAXINTEGER = 1<<63 - 1
	LUA_MININTEGER = -1 << 63
)

/* basic types */
const (
	LUA_TNONE = iota - 1 // -1
	LUA_TNIL
	LUA_TBOOLEAN
	LUA_TLIGHTUSERDATA
	LUA_TNUMBER
	LUA_TSTRING
	LUA_TTABLE
	LUA_TFUNCTION
	LUA_TUSERDATA
	LUA_TTHREAD
)

/* arithmetic functions */
const (
	LUA_OPADD  = iota // +
	LUA_OPSUB         // -
	LUA_OPMUL         // *
	LUA_OPMOD         // %
	LUA_OPPOW         // ^
	LUA_OPDIV         // /
	LUA_OPIDIV        // //
	LUA_OPBAND        // &
	LUA_OPBOR         // |
	LUA_OPBXOR        // ~
	LUA_OPSHL         // <<
	LUA_OPSHR         // >>
	LUA_OPUNM         // -
	LUA_OPBNOT        // ~
)

/* comparison functions */
const (
	LUA_OPEQ = iota // ==
	LUA_OPLT        // <
	LUA_OPLE        // <=
)

/* thread status */
const (
	LUA_OK = iota
	LUA_YIELD
	LUA_ERRRUN
	LUA_ERRSYNTAX
	LUA_ERRMEM
	LUA_ERRGCMM
	LUA_ERRERR
	LUA_ERRFILE
)

File: code/rust/ch08/src/api/consts.rs
/* basic types */
pub const LUA_TNONE: i8 = -1;
pub const LUA_TNIL: i8 = 0;
pub const LUA_TBOOLEAN: i8 = 1;
pub const LUA_TLIGHTUSERDATA: i8 = 2;
pub const LUA_TNUMBER: i8 = 3;
pub const LUA_TSTRING: i8 = 4;
pub const LUA_TTABLE: i8 = 5;
pub const LUA_TFUNCTION: i8 = 6;
pub const LUA_TUSERDATA: i8 = 7;
pub const LUA_TTHREAD: i8 = 8;

/* arithmetic functions */
pub const LUA_OPADD: u8 = 0; // +
pub const LUA_OPSUB: u8 = 1; // -
pub const LUA_OPMUL: u8 = 2; // *
pub const LUA_OPMOD: u8 = 3; // %
pub const LUA_OPPOW: u8 = 4; // ^
pub const LUA_OPDIV: u8 = 5; // /
pub const LUA_OPIDIV: u8 = 6; // //
pub const LUA_OPBAND: u8 = 7; // &
pub const LUA_OPBOR: u8 = 8; // |
pub const LUA_OPBXOR: u8 = 9; // ~
pub const LUA_OPSHL: u8 = 10; // <<
pub const LUA_OPSHR: u8 = 11; // >>
pub const LUA_OPUNM: u8 = 12; // -
pub const LUA_OPBNOT: u8 = 13; // ~

/* comparison functions */
pub const LUA_OPEQ: u8 = 0; // ==
pub const LUA_OPLT: u8 = 1; // <
pub const LUA_OPLE: u8 = 2; // <=
